src/wallet/rpcwallet.cpp

```diff
@@ -221,17 +221,69 @@ void TxExpandedToJSON(const CWalletTransactionBase& tx,  UniValue& entry)
     }
 }
 
-void WalletTxToJSON(const CWalletTransactionBase& wtx, UniValue& entry, isminefilter filter)
+static int getCertMaturityHeight(const CWalletTransactionBase& wtx)
+{
+    if (wtx.hashBlock.IsNull())
+    {
+        // this is the case when wtx has not yet been mined (zero conf)
+        return -1;
+    }
+
+    int matDepth = wtx.bwtMaturityDepth;
+
+    // get index of the block which containn this cert
+    CBlockIndex *pindex = nullptr;
+    BlockMap::iterator it = mapBlockIndex.find(wtx.hashBlock);
+    if (it == mapBlockIndex.end())
+    {
+        throw JSONRPCError(RPC_TYPE_ERROR, strprintf(
+            "coluld not find cert maturity height since block %s is not in active chain",
+            wtx.hashBlock.ToString()));
+    }
+
+    return it->second->nHeight + matDepth;
+}
+
+void WalletTxToJSON(const CWalletTransactionBase& wtx, UniValue& entry, isminefilter filter, bool isCertMaturingInRange = false)
```

Comment:
Can we add a comment to explain that the optional parameter `isCertMaturingInRange` is only used when the function is called by `listsinceblock()` -> `ListTransactions()`?

Otherwise, looking at this function only, it is not clear to me why we need such a parameter.